File: Placements.DataAccess/placement/Models/Calendarevent.cs
using System;
using System.Collections.Generic;

namespace Placements.DataAccess.Placement.Models;

public partial class Calendarevent
{
    public long Id { get; set; }

    public DateTime? EventStartDateTime { get; set; }

    public DateTime? EventEndDateTime { get; set; }

    public string? EventType { get; set; }

    public string? EventDescription { get; set; }

    public long? OrgId { get; set; }

    public long? CompanyId { get; set; }

    public ulong IsDeleted { get; set; }

    public virtual Companydatum? Company { get; set; }

    public virtual ICollection<Jobinterviewround> Jobinterviewrounds { get; set; } = new List<Jobinterviewround>();

    public virtual Campusregistration? Org { get; set; }
}

File: web-app/src/app/services/types/Calendarevent.ts
import { Jobinterviewround } from "./Jobinterviewround";

export type Calendarevent = {
  Id: number;
  EventStartDateTime: Date;
  EventEndDateTime: Date;
  EventType: string;
  EventDescription: string;
  OrgId: number;
  CompanyId: number;
  IsDeleted: boolean;
  Jobinterviewrounds: Jobinterviewround[];
};

export type PostCalendarevent = Calendarevent | {};

export type PostCalEvent = {
  Id: number;
  EventStartDateTime: Date;
  EventEndDateTime: Date;
  EventType: string;
  EventDescription: string;
  OrgId: number;
  CompanyId: number;
  IsDeleted: boolean;
  Jobinterviewrounds: { eventId?: number };
};
